app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -85,32 +113,20 @@ class CreditCardsSettingFragment : PreferenceFragmentCompat() {
                 .getString(R.string.preferences_credit_cards_sync_cards_across_devices),
             loggedInTitle = requireContext()
                 .getString(R.string.preferences_credit_cards_sync_cards),
-            onSignInToSyncClicked = {
-                val directions =
-                    CreditCardsSettingFragmentDirections.actionCreditCardsSettingFragmentToTurnOnSyncFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
-            },
-            onReconnectClicked = {
-                val directions =
-                    CreditCardsSettingFragmentDirections.actionGlobalAccountProblemFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
-            }
+            onSignInToSyncClicked = { navigateToSyncFragment() },
+            onReconnectClicked = { navigateToAccountProblemFragment() }
         )
+
+        togglePrefsEnabledWhileAuthenticating(creditCardPreferences, true)
     }
 
     override fun onPreferenceTreeClick(preference: Preference): Boolean {
         when (preference.key) {
             getPreferenceKey(R.string.pref_key_credit_cards_add_credit_card) -> {
-                val directions =
-                    CreditCardsSettingFragmentDirections
-                        .actionCreditCardsSettingFragmentToCreditCardEditorFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
+                navigateToCreditCardEditorFragment()
             }
             getPreferenceKey(R.string.pref_key_credit_cards_manage_saved_cards) -> {
-                val directions =
-                    CreditCardsSettingFragmentDirections
-                        .actionCreditCardsSettingFragmentToCreditCardsManagementFragment()
-                findNavController().navigateBlockingForAsyncNavGraph(directions)
+                verifyCredentialsOrShowSetupWarning(preference.context, creditCardPreferences)
```

Comment:
Just a little drive-by comment, but I figured I'll chime in...

There's no reason to require authentication to display a list of cards, or even to open a specific card. Sensitive information (card number) should not be displayed right away. We can't actually display it - it's encrypted! So, that's what I imagined would be guarded by an auth prompt - the action to display a specific credit card number.

To put it in other words - there's a cryptographic key that we need to fetch in order to decrypt or encrypt sensitive information related to a credit card number. The act of obtaining a key, while not technically necessary, should be protected by some authentication mechanism (e.g. a biometrics prompt).

What I think is happening in this PR is more akin to how we're treating logins currently. That's not wrong, of course. However, even for logins we'll be soon switching to a per-field authentication model (vs per-db, as it is currently), which would mean that only username/password fields will be encrypted, while the rest will be in plaintext. That'll allow us to follow the same pattern for both credit cards and logins at the product UX level (e.g. require auth only when accessing sensitive information).

Reply:
Thanks for the context! We should probably check in with UX on this because AFAIK this is the user authentication flow that they want (for now)